### PR TITLE
Make integration tests not depend on e2e tests

### LINKED ISSUE #100456: Integration tests should not depends on e2e tests
Originally tracked at #89185. After fixing "not include test files in import-boss" in #90380, we have a regression here:
`test/integration/auth/bootstraptoken_test.go` is using `k8s.io/kubernetes/test/e2e/lifecycle/bootstrap`.

We need remove dependencies from integration tests to e2e tests again.

/assign
/sig testing

### PR DESCRIPTION
**Pull request by gavinfish**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

We need remove dependencies from integration tests to e2e tests.

#### Which issue(s) this PR fixes:

Fixes #100456

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/auth/bootstraptoken_test.go [hunk 1]
```diff
@@ -47,23 +46,17 @@ func (b bootstrapSecrets) Get(name string) (*corev1.Secret, error) {
...
 // TestBootstrapTokenAuth tests the bootstrap token auth provider
 func TestBootstrapTokenAuth(t *testing.T) {
-	tokenID, err := bootstraputil.GenerateTokenID()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	secret, err := bootstraputil.GenerateTokenSecret()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
```
Comment (neolit123) [inline test/integration/auth/bootstraptoken_test.go:57]: given GenerateTokenID and GenerateTokenSecret are already tested elsewhere.

/sig auth
/lgtm

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/milestone v1.21

**Comment by spiffxp** [comment]
/triage accepted
/priority backlog
I see minimal potential disruption here but marking low priority in case we end up not having the bandwidth to deal with this

**Comment by spiffxp** [comment]
It appears the test this PR touches is failing. If not resolved today, suggest we punt to v1.22

**Pull request merged**